### ISSUE TITLE
Reserve k8s master Public IP, to preserve over on instance reboots

### ISSUE THREAD

**Issue by amirhmoradi** [hobby-kube/provisioning#6]
Hi & thanks for the great work 
I would like to be able to reserver the public IP for k8s master (host 0), on scaleway, because it might change on host reboot otherwise.

I tried creating an IP resource, but can't get around dependencies in terraform.
The issue is that one the host 0 is created, other resources start creating (etcd, fw, dns...) and in scaleway it seems to reserve the IP address after instance creation, so already launched creating services take the first public IP, and not the reserved one.

ex:
in provider/scaleway/main.tf I added:

```
resource "scaleway_ip" "master_ip" {
  server = "${scaleway_server.host.0.id}"
}
```

and modified the public_ips output like so:

```
output "public_ips" {
  depends_on = ["scaleway_ip.master_ip.ip"]
  value      = ["${scaleway_server.host.*.public_ip}"]
}
```

What happens is:
```
module.firewall.null_resource.firewall.0 (remote-exec): Connecting to remote host via SSH...
module.firewall.null_resource.firewall.0 (remote-exec):   Host: <FIRST_RANDOM_PUBLIC_IP>
module.firewall.null_resource.firewall.0 (remote-exec):   User: root
module.firewall.null_resource.firewall.0 (remote-exec):   Password: false
module.firewall.null_resource.firewall.0 (remote-exec):   Private key: false
module.firewall.null_resource.firewall.0 (remote-exec):   SSH Agent: true
```
...
...
```
Error applying plan:
* timeout
* module.firewall.null_resource.firewall[0]: 1 error(s) occurred:
```

And I noted that the DNS provider also takes and sets up the domain A record with `<FIRST_RANDOM_PUBLIC_IP>`.

Once the instance is created on scaleway, the `<FIRST_RANDOM_PUBLIC_IP>` changes and the instance is given a reserved public IP.

Any ideas on this?

**Comment by pstadler** [comment]
First of all, I never had an IP changing on reboot.

When I first experimented with with the Scaleway provider, I ran into similar issues when using the `scaleway_ip` resource. `depends_on` is currently pretty flaky and there's an [ongoing discussion](https://github.com/hashicorp/terraform/issues/10462) around implementing this feature on a module basis.

Besides that, your `public_ips` output should probably consist of master_ip, public_ip_host2, public_ip_host3.
